### PR TITLE
Support the new Databricks Unity V2 Proxy

### PR DESCRIPTION
**Pull request by wjohnson**
The Databricks Unity V2 Handler is a copy paste of the DatabricksDeltaHandler but uses the unity class name.

I believe it's worth separating out because we have no idea what Unity will turn into and it should be handled independently from Delta.

Signed-off-by: Will Johnson <will@willj.co>

### Problem

Starting with Databricks Runtime 10.4, Databricks introduced a UnityV2Proxy class that sits in front of the normal hive metastore. We now have to handle this new catalog layer.

Closes: #964 

### Solution

The Databricks Unity V2 Handler is a copy paste of the DatabricksDeltaHandler but uses the unity class name.

I believe it's worth separating out because we have no idea what Unity will turn into and it should be handled independently from Delta.

- [ ] Your change modifies the [core](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json) OpenLineage model
- [ ] Your change modifies one or more OpenLineage [facets](https://github.com/OpenLineage/OpenLineage/tree/main/spec/facets)

If you're contributing a new integration, please specify the scope of the integration and how/where it has been tested (e.g., Apache Spark integration supports `S3` and `GCS` filesystem operations, tested with AWS EMR).

### Checklist

- [X] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [X] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [ ] Your changes are accompanied by tests (_if relevant_)
- [X] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [ ] You've updated the [`CHANGELOG.md`](https://github.com/OpenLineage/OpenLineage/blob/main/CHANGELOG.md) with details about your change under the "Unreleased" section (_if relevant, depending on the change, this may not be necessary_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [ ] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

### CODE REVIEW

File: integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/lifecycle/plan/catalog/DatabricksUnityV2Handler.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  }
+
+  @Override
+  public boolean isClass(TableCatalog tableCatalog) {
+    return "com.databricks.sql.managedcatalog.UnityCatalogV2Proxy"
+        .equals(tableCatalog.getClass().getCanonicalName());
+  }
+
+  @Override
+  public DatasetIdentifier getDatasetIdentifier(
```
Comment (pawel-big-lebowski) [inline integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/lifecycle/plan/catalog/DatabricksUnityV2Handler.java:60]: If it is a copy/paste from `DatabricksDeltaHandler` (as described in PR), please consider creating an `AbstractDatabricksHandler` class with `isClass` and `getDatasetIdentifier` methods implemented and let classes `DatabricksUnityV2Handler` and `DatabricsDeltaHandler` extend this abstract class.

Comment (wjohnson) [reply integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/lifecycle/plan/catalog/DatabricksUnityV2Handler.java:60]: @pawel-big-lebowski can do.  My only concern is that we have no idea how Unity is going to evolve.  Is it still worth handling if Unity suddenly grows to be a completely different beast than the way Databricks Delta catalog was being handled?

Comment (pawel-big-lebowski) [reply integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/lifecycle/plan/catalog/DatabricksUnityV2Handler.java:60]: We make use of non-public API. There is no guarantee on how do they evolve and there is little we can do about that. For the future versions we can create new Handlers within `spark32`, `spark33`.... projects.

Comment (wjohnson) [reply integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/lifecycle/plan/catalog/DatabricksUnityV2Handler.java:60]: Got it!  Thank you so much for the guidance, Pawel!  I really appreciate the perspective and assurance that we can handle this as they grow.

Comment (wjohnson) [reply integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/lifecycle/plan/catalog/DatabricksUnityV2Handler.java]: @pawel-big-lebowski  per your suggestion, I've refactored this to use an `AbstractDatabricksHandler` instead and noted their relationships.

File: integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/lifecycle/plan/catalog/AbstractDatabricksHandler.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+/**
+ * The AbstractDatabricksHandler is intended to support Databricks' custom proxies which supersede
+ * the open source Delta Catalog. The proprietary class name of
+ * com.databricks.sql.transaction.tahoe.catalog.DeltaCatalog is used by Databricks instead of the
+ * open source class name of org.apache.spark.sql.delta.catalog.DeltaCatalog. It is used in the same
+ * way as the {@link DeltaHandler}.
+ */
+@Slf4j
+public abstract class AbstractDatabricksHandler implements CatalogHandler {
+  final String DATABRICKS_CLASS_NAME_STRING;
```
Comment (pawel-big-lebowski) [inline integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/lifecycle/plan/catalog/AbstractDatabricksHandler.java:33]: Uppercase with underscore is a naming convention for constants in Java, while `DATABRICKS_CLASS_NAME_STRING` is not such. Consider renaming it to `databricksClassNameString`.

**Review by pawel-big-lebowski — APPROVED** [review]
LGTM 💯 👍 
A single minor comment added.

File: integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/lifecycle/plan/catalog/AbstractDatabricksHandler.java (hunk 2 above)
Comment (wjohnson) [reply integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/lifecycle/plan/catalog/AbstractDatabricksHandler.java]: Changes made!  Thank you so much, for the excellent guidance as usual, Pawel!

**Pull request merged**